visma/functions/constant.py

```diff
@@ -33,13 +39,192 @@ def differentiate(self):
+                result = Constant()
+                if self.value == 0:
+                    if self.power == 0:
+                        self.value = 1
+                        self.power = 1
+                        result = Constant()
+                        result.scope = self.scope
+                        result.power = self.power
+                        result.value = self.calculate()
+                else:
+                    result = Constant()
+                    result.scope = self.scope
+                    result.power = self.power
+                    result.value = self.calculate()
+                return result
+        elif self.isZero():
+            return other
+        elif other.isZero():
+            return self
+        elif isinstance(other, Expression):
+            if other.power == 1:
+                constFound = False
+                for i, var in enumerate(other.tokens):
+                    if isinstance(var, Constant):
+                        other.tokens[i] = self + var
+                        constFound = True
+                        break
+                if not constFound:
+                    other.tokens.extend([Plus(), self])
+                return other
+            else:
+                pass
+        self.value = self.calculate()
+        self.power = 1
+        self.coefficient = 1
+        exprAdd = Expression([self, Plus(), other])     # Make an Expression and assign the Tokens attribute with the Constant and the Other Variable, Trig. function,...etc.
+        return exprAdd
+
+    def __rsub__(self, other):
+        return Constant(0) - self + other
+
+    def __sub__(self, other):
+        if isinstance(other, Constant):
+            self = self + Constant(-1, 1, 1) * other
+            return self
+        elif isinstance(other, Expression):
+            expression = Expression()
+            expression.tokens = [self]
+            if other.power == 1:
+                coeff = other.coefficient
+                for i, token in enumerate(other.tokens):
+                    print(expression, " ", type(token), other.tokens[i-1])
+                    if isinstance(token, Constant):
+                        if other.tokens[i-1].value == '+' or i == 0:
+                            expression.tokens[0] = Constant(self.calculate() - token.calculate()*coeff)
+                        elif other.tokens[i-1].value == '-':
+                            expression.tokens[0] = Constant(self.calculate() + token.calculate()*coeff)
+                            print(expression.tokens[0])
+                    elif isinstance(token, Variable):
+                        if other.tokens[i-1].value == '+' or i == 0:
+                            expression.tokens.extend([Minus(), Variable(token)])
+                        elif other.tokens[i-1].value == '-':
+                            expression.tokens.extend([Plus(), Variable(token)])
+                self.type = 'Expression'
```

Comment:
Updating `self.type` is not required, we can directly make use of class type.  The `type` property can be removed from all classes.

Reply:
done and removed `type` property from all the Classes